OgreMain/src/OgreSceneNode.cpp

```diff
@@ -315,6 +324,49 @@ namespace Ogre {
         mChildren.clear();
         needUpdate();
     }
+    //-----------------------------------------------------------------------
+    void SceneNode::destroyChildGraphSegment(const String& name) {
+        SceneNode* pChild = static_cast<SceneNode*>(getChild(name));
+        pChild->destroyGraphSegment();
+
+        removeChild(name);
+        pChild->getCreator()->destroySceneNode(name);
+
+    }
+
+    void SceneNode::destroyChildGraphSegment(unsigned short index) {
+        SceneNode* pChild = static_cast<SceneNode*>(getChildren()[index]);
```

Comment:
```suggestion
        SceneNode* pChild = static_cast<SceneNode*>(removeChild(index));
```
otherwise pChild might be OOB. This is an issue in the original code as well